libraries/plugins/mongo_db/mongo_db_writer.cpp

```diff
@@ -0,0 +1,71 @@
+#include <golos/plugins/mongo_db/mongo_db_writer.hpp>
+#include <fc/log/logger.hpp>
+#include <mongocxx/exception/exception.hpp>
+
+
+namespace golos {
+namespace plugins {
+namespace mongo_db {
+
+    using namespace bsoncxx::types;
+    using namespace bsoncxx::builder;
+    using bsoncxx::builder::basic::kvp;
+
+    const std::string mongo_db_writer::blocks_col = "Blocks";
+    const std::string mongo_db_writer::trans_col = "Transactions";
+
+    void mongo_db_writer::initialize(const std::string& uri_str) {
+
+        try {
+            uri = mongocxx::uri {uri_str};
+            mongo_conn = mongocxx::client {uri};
+            db_name = uri.database().empty() ? "Golos" : uri.database();
+            ilog("MongoDB plugin initialized.");
+        }
+        catch (mongocxx::exception & ex) {
+            ilog(ex.what());
+            throw;
```

Comment:
Don't throw an unknown for golosd exception.
If you need throw an exception - declare new type based on fc::exception and throw it.

Reply:
You right. In fact, i guess if mongo plugin would be broken its not the end of the world, we can just log the exception and get on with our lives.